design/ceph-mixins.md

```diff
@@ -0,0 +1,61 @@
+# Monitoring ceph storage with ceph-mixins
+
+---
+
+## Overview
+
+Currently, rook deploys ceph storage but not ceph monitoring. It can be deployed manually using documented steps. But still, it lacks Prometheus Alerts and Recording Rules that is useful in easy monitoring of ceph storage.
+
+This is where **ceph-mixins** comes in. Ceph-mixins defines a minimalistic approach to package together the ceph storage specific prometheus alerts, recording rules and grafana dashboards, in a simple and platform agnostic way.
+
+Rook can use ceph-mixins as the default solution for monitoring ceph storage.
+
+## Design
+
+---
+
+### Responsibilities
+
+1. Getting the latest updates from ceph-mixins into rook.
+2. Building ceph-mixins artifacts into kubernetes custom resources.
+3. Deploying prometheus bundle generated by ceph-mixins.
+
+### Deployment
+
+1. Fetching latest resources from ceph-mixins
+
+   Ceph-mixins uses Jsonnet for writing very flexible and extensible configurations pertaining to prometheus alerts, recording rules and grafana dashboards.
+
+   These configuration files are bundled together and managed using jsonnet-bundler.
+
+    **Changes required in Rook**
+
+    A new `jsonnet` directory should be created in Rook which will hold configurations which will be used to pull resources from ceph-mixins. These configurations will allow rook to version lock the resources being pulled and extend or modify it.
+
+2. Building deployment assets
+
+   A build script will be used to generate yaml files that can be applied to any kubernetes cluster.
+
+    **Changes required in Rook**
+
+    A new `assets` directory should be created in Rook which will hold all the manifests that can be applied to the kubernetes cluster.
+
+3. Deploying
+
+   1. In Kubernetes
+
+        `kubectl create -f assets/`
+
+        This will deploy Prometheus Operator, Prometheus instance, PrometheusRules (which contains alerts and recording rules), and Grafana with dashboards.
```

Comment:
Since the creation of this [monitoring help topic](https://rook.io/docs/rook/v0.9/ceph-monitoring.html), the approach has been that prometheus is an independent operator and rook should not own yaml files to install and configure prometheus. With this approach to create all the assets, you're saying that we would commit a yaml file in the rook repo that starts the prometheus operator? This seems outside the scope of rook.

Reply:
Initially, the idea was to provide an entire monitoring stack as an addon to rook. But, if it is out of scope of rook, we will limit it to the bare minimum i.e one `prometheus-rules.yml` file.